experimental/Schemes/CartierDivisor.jl

```diff
@@ -174,34 +174,60 @@ function cartier_divisor(IP::AbsProjectiveScheme, f::Union{MPolyDecRingElem, MPo
   return one(ZZ)*effective_cartier_divisor(IP, f)
 end
 
+### Decompostion of an effective Cartier Divisor into irreducible components
+### (specialized variant of associated_points, using pure codimension 1
+###  and taking multiplicities into account)
+@doc raw"""
+    irreducible_decomposition(C::EffectiveCartierDivisor)
+
+Return a `Vector` of pairs ``(I,k)`` corresponding to the irreducible components of ``C``. More precisely,  each ``I`` is a prime  `IdealSheaf` corresponding to an irreducible component of ``C`` and ``k``is the multiplicity of this component in ``C``.
+"""
+function irreducible_decomposition(C::EffectiveCartierDivisor)
+  cov = trivializing_covering(C)
+  X = scheme(C)
+  OOX = OO(X)
+
+  charts_todo = copy(patches(cov))
+  I = ideal_sheaf(C)
+  associated_primes_temp = Vector{Tuple{IdealSheaf, Int}}()  ## already identified components
+
+  # run through all charts and collect further irreducible components
+  while length(charts_todo) > 0
+    U = pop!(charts_todo)
+    !is_one(I(U)) || continue                                ## supp(C) might not meet all charts
+    I_temp=I(U)
+
+    for (J,_) in associated_primes_temp
+      !is_one(J(U)) || continue
+      I_temp=saturation(I_temp,J(U))                         ## kick out known components
+      !is_one(I_temp) || break                               ## break if nothing left
+    end
+
+    !is_one(I_temp) || break                                 ## break if nothing left
+    components_here = minimal_primes(I(U))
```

Comment:
Wouldn't it suffice to take the minimal primes of `I_temp`? The other components are already taken care of, aren't they?

Reply:
Indeed. This is a simple typo (no changed back after some testing)